File: SMI-SDK/Java SDK/src/pt/sapo/sdb/trace/Trace.java
package pt.sapo.sdb.trace;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Trace {    
    
    public Authentication authentication;
    public Date start;
    public Date end;
    public Integer tenantId;
    public String traceId ;
    public String id ;
    public String activityId ;
    public String parentId ;
    
    public Server Server = null;
    public Client Client = null;
    public Service Service = null;
    public Summary Summary = null;
                
    public List<Entry> entries = new ArrayList<Entry>();
    public ArrayList<Message> messages = new ArrayList<Message>();
}

File: SMI-SDK/.NET SDK/Library/sdb/smi/Trace.cs
using System;
using System.Collections.Generic;
namespace pt.sapo.sdb.smi
{
    public class Trace
    {
        public Authentication Authentication { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public int TenantId { get; set; }
        public String TraceId { get; set; }
        public String Id { get; set; }
        public String ActivityId { get; set; }
        public String ParentId { get; set; }

        public Server Server { get; set; }
        public Client Client { get; set; }
        public Service Service { get; set; }
        public Summary Summary { get; set; }

        public ICollection<Entry> Entries { get; protected set; }
        public IList<Message> Messages { get; protected set; }

        public Trace() {
            Entries = new List<Entry>();
            Messages = new List<Message>();
        }
    }
}